src/odemis/acq/stream/_base.py

```diff
@@ -787,15 +832,185 @@ def _image_thread(wstream):
+            last execution of _updateImage
+        prev_proj_cache (dictionary): projected tiles cache from the
+            last execution of _updateImage
+        return (tuple(DataArray, DataArray)): raw tile and projected tile
+        """
+        # the key of the tile on the cache
+        tile_key = "%d-%d-%d" % (x, y, z)
+
+        # if the tile has been already cached, read it from the cache
+        if tile_key in prev_proj_cache:
+            proj_tile = prev_proj_cache[tile_key]
+            raw_tile = prev_raw_cache[tile_key]
+        elif tile_key in self._projectedTilesCache:
+            proj_tile = self._projectedTilesCache[tile_key]
+            raw_tile = self._rawTilesCache[tile_key]
+        else:
+            # The tile was not cached, so it must be read from the file
+            raw_tile = self._das.getTile(x, y, z)
+            proj_tile = self._projectTile(raw_tile)
+
+        # cache raw and projected tiles
+        self._rawTilesCache[tile_key] = raw_tile
+        self._projectedTilesCache[tile_key] = proj_tile
+        return (raw_tile, proj_tile)
+
+    def _projectTile(self, tile):
+        """
+        Project the tile
+        tile (DataArray): Raw tile
+        return (DataArray): Projected tile
+        """
+        return self._projectXY2RGB(tile, self.tint.value)
+
+    def _getTilesFromSelectedArea(self):
+        """
+        Get the tiles inside the region defined by .rect and .mpp
+        return ((DataArray, DataArray)): Raw tiles and projected tiles
+        """
+        z = self._zFromMpp()
+        rect = self._rectWorldToPixel(self.rect.value)
+        # convert the rect coords to tile indexes
+        rect = [l / (2 ** z) for l in rect]
+        rect = [int(math.floor(l / self._das.tile_shape[0])) for l in rect]
+        x1, y1, x2, y2 = rect
+        curr_mpp = self.mpp.value
+        curr_rect = self.rect.value
+        # store the previous cache to use in this execution
+        prev_raw_cache = self._rawTilesCache
+        prev_proj_cache = self._projectedTilesCache
+        # empty current caches
+        self._rawTilesCache = {}
+        self._projectedTilesCache = {}
+        # Execute at least once. If mpp and rect changed in
+        # the last execution of the loops, execute again
+        mpp_rect_changed = True
+        while mpp_rect_changed:
+            raw_tiles = []
+            projected_tiles = []
+            mpp_rect_changed = False
+            for x in range(x1, x2 + 1):
+                rt_column = []
+                pt_column = []
+                # check if .mpp or .rect changed in the middle of the process.
+                # it is common to happen when .rect and .mpp are changed simultaneously
+                if curr_mpp != self.mpp.value or curr_rect != self.rect.value:
+                    curr_mpp = self.mpp.value
+                    curr_rect = self.rect.value
+                    # the 4 lines below avoids that lots of old tiles
```

Comment:
You can move the .update + chache={} at the beginning of the `while` loop, and this avoids doing it once before the `while` loop. (Initialise prev_* with {}).